### PR TITLE
Feature/prlc100 20/add people

### PR DESCRIPTION
**Pull request by hmcts-abduldaim**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRLC100-20

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/common/form/fields/link.njk [hunk 1]
```diff
@@ -1 +1 @@
- <a class="govuk-link" href="{{getContent(field.link)}}" target ="_blank">{{getContent(field.label)}}</a>
\ No newline at end of file
+ <a class="{{field.classes}}" href="{{field.href}}" >{{getContent(field.label)}}</a>
```
Comment (yogendraU) [inline src/main/steps/common/form/fields/link.njk:1]: any specific reason for removing target?

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/app/case/definition.ts [hunk 2]
```diff
@@ -313,6 +313,15 @@ export interface PlacementOrder {
   placementOrderDate?: CaseDate | string;
 }
 
+
+export type C100Applicant = {
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/case/definition.ts:317]: Applicant already exists and this is why had to change it to C100Applicant

File: src/main/steps/common/form/fields/link.njk (hunk 1 above)
Comment (hmcts-abduldaim) [reply src/main/steps/common/form/fields/link.njk:1]: target allows to open it in another tab. In future if such need is required we can create a target link

File: src/main/app/case/case.ts [hunk 3]
```diff
@@ -105,6 +114,7 @@ export interface Case {
   injuctiveOrder?: orderInterface;
   underTakingOrderOptions?: YesOrNo;
   underTakingOrder?: orderInterface;
+  allApplicants?: C100ListOfApplicants;
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/case/case.ts]: the change will require refactoring of a large scale

File: src/main/app/case/case.ts [hunk 4]
```diff
@@ -1,7 +1,16 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { AnyObject } from '../controller/PostController';
 
-import { CaseData, ContactDetails, OtherName, OtherProceedings, State, YesOrNo, orderInterface } from './definition';
+import {
+  C100ListOfApplicants,
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/case/case.ts:5]: this is used as type for just c100 application

**Review by hmcts-abduldaim — COMMENTED** [review]
please review the comments

File: sonar-project.properties [hunk 5]
```diff
@@ -6,7 +6,7 @@ sonar.tests=src/test/
...
+sonar.exclusions=src/**/*.test.ts,**/__mocks__/**,src/**/*content.ts,src/**/components/*.ts,src/**/definition.ts,src/**/summary/util.ts,src/**/summary/util.ts,src/**/steps/constants.ts,src/**/willingness-to-attend-miam/miam-cost-exemptions.ts,src/**/steps/index.ts,src/**/steps/urls.ts,src/**/case/CaseApi.ts,src/**/app/controller/PostController.ts,src/**/checkanswers/ConfirmContactDetailsGetController.ts,src/**/respondent/task-list/utils.ts,src/**/respondent/respondentcaseSequence.ts,src/main/steps/common/summary/utils.ts,src/**/applicant/task-list/utils.ts,src/**/components/common/address-history.ts,src/**/ca-da-respondent/task-list/utils.ts,src/main/steps/c100-rebuild/c100sequence.ts,src/main/app/fees/fees-lookup-api.ts,src/main/steps/c100-rebuild/child-details/**/*.ts,src/main/steps/c100-rebuild/other-proceedings/**/*.ts,src/main/steps/c100-rebuild/applicant/**/*.ts
```
Comment (yogendraU) [inline sonar-project.properties:9]: any reason why applicant folder is added as exclusion

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/applicant/add-applicants/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+
+export type AnyProperty = any;
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/c100-rebuild/applicant/add-applicants/index.d.ts]: it's been fixed and pushed

File: src/main/steps/c100-rebuild/applicant/add-applicants/getController.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      additionalData: {
+        req,
+      },
+    });
+
+    const sessionErrors = req.session?.errors || [];
+    if (req.session?.errors) {
+      req.session.errors = undefined;
+    }
+    super.clearConfidentialitySessionSaveData(req);
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/c100-rebuild/applicant/add-applicants/getController.ts]: it's been fixed and pushed

File: sonar-project.properties (hunk 5 above)
Comment (hmcts-abduldaim) [reply sonar-project.properties]: it's been fixed and pushed

**Review by yogendraU — APPROVED** [review]

**Pull request merged**